pages/api/rss.ts

```diff
@@ -0,0 +1,23 @@
+import { NextRequest } from 'next/server';
+import { allBlogs, allAuthors } from 'contentlayer/generated';
+import { generateRSS } from '../../lib/utils/generate-rss';
+import { siteMetadata } from '../../data/siteMetadata';
+
+const handler = async (req: NextRequest) => {
+  const { searchParams } = new URL(req.url);
+  const tag = searchParams.get('tag');
+  const feedUrl = tag ? `tags/${tag}/feed.xml` : 'feed.xml';
+  const feed = await generateRSS(siteMetadata, allBlogs, allAuthors, tag, feedUrl);
+  return new Response(feed, {
+    headers: {
+      'Content-Type': 'application/xml',
+      'Cache-Control': 'public, s-maxage=1200, stale-while-revalidate=600',
```

Comment:
These cache values may need to be changed depending on what makes sense for the site.

Reply:
This seems sensible to me.